test/client_test.py

```diff
@@ -277,3 +277,13 @@ def get_assignments(test_case):
         )
         for subject in test_case.get("subjectsWithAttributes", [])
     ]
+
+@pytest.mark.parametrize("test_case", test_data)
+def test_get_numeric_assignment_on_bool_feature_flag_should_return_none(test_case):
```

Comment:
Wrote this test to confirm that this fixes the issue. Before the code change above, this test failed

```
FAILED test/client_test.py::test_get_numeric_assignment_on_bool_feature_flag_should_return_none[test_case0] - assert [True, False, False, False] == [None, None, None, None]
```